plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -442,8 +442,8 @@ public TimeBoundary(
         {
             if (timeColumn != null && timeValue != null) {
                 // See org.apache.pinot.broker.requesthandler.BaseBrokerRequestHandler::attachTimeBoundary
-                offlineTimePredicate = Optional.of(format("%s <= %s", timeColumn, timeValue));
-                onlineTimePredicate = Optional.of(format("%s > %s", timeColumn, timeValue));
+                offlineTimePredicate = Optional.of(format("%s <= '%s'", timeColumn, timeValue));
```

Comment:
Note: this change also works with pinot 1.1.0. It was necessary as the type of the time boundary value changed from long to timestamp in pinot 1.2.0 (i.e. a timestamp literal), so single quotes are needed.

This works for both timestamp and long types now.